ports/pulsar-client-cpp/unofficial-pulsar-config.cmake

```
@@ -0,0 +1,58 @@
+if (NOT TARGET unofficial::pulsar::pulsar)
+    get_filename_component(VCPKG_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
+    get_filename_component(VCPKG_IMPORT_PREFIX "${VCPKG_IMPORT_PREFIX}" PATH)
+    get_filename_component(VCPKG_IMPORT_PREFIX "${VCPKG_IMPORT_PREFIX}" PATH)
+
+    find_path(_pulsar_include_dir NAMES "pulsar/Client.h" PATH ${VCPKG_IMPORT_PREFIX})
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(VCPKG_IMPORT_PREFIX "${VCPKG_IMPORT_PREFIX}/debug")
+    endif ()
+    if ("@VCPKG_LIBRARY_LINKAGE@" STREQUAL "dynamic")
+        find_library(_pulsar_library NAMES libpulsar.so libpulsar.dylib pulsar.lib PATH ${VCPKG_IMPORT_PREFIX})
+        set(_pulsar_link_static_library OFF)
+        message(STATUS "Found dynamic _pulsar_library: ${_pulsar_library}")
+    else ()
+        find_library(_pulsar_library NAMES libpulsar.a pulsar-static.lib PATH ${VCPKG_IMPORT_PREFIX})
+        set(_pulsar_link_static_library ON)
+        message(STATUS "Found static _pulsar_library: ${_pulsar_library}")
+    endif ()
+    if (NOT _pulsar_include_dir OR NOT _pulsar_library)
+        message(FATAL_ERROR "Broken installation of vcpkg port pulsar-client-cpp")
+    endif ()
+
+    include(CMakeFindDependencyMacro)
+    find_dependency(OpenSSL)
+    find_dependency(ZLIB)
+    find_dependency(protobuf CONFIG)
+    find_dependency(CURL CONFIG)
+    find_dependency(zstd CONFIG)
+    find_dependency(snappy CONFIG)
+    if (MSVC)
+        find_dependency(dlfcn-win32 CONFIG)
+    endif ()
+
+    add_library(unofficial::pulsar::pulsar INTERFACE IMPORTED)
```

```suggestion
    add_library(unofficial::pulsar::pulsar UNKNOWN IMPORTED)
```
Interface libs don't have a location.